examples/yacat/yacat.py

```diff
@@ -0,0 +1,163 @@
+            yield ctx.commit()
+            task.accept_task(result=output_file)
+
+    async def worker_find_password(ctx: WorkContext, tasks):
+        in_hash_filename = str(script_dir / "in.hash")
+        ctx.send_file(in_hash_filename, "/golem/work/in.hash")
+        ctx.send_file("empty.txt", "/golem/work/hashcat.potfile")
+
+        async for task in tasks:
+            skip = task.data
+            limit = skip + step
+            ctx.begin()
+            ctx.run(
+                f"/bin/sh",
+                "--",
+                "-c",
+               # "touch /golem/work/hashcat.potfile;",
+                f"hashcat -a 3 -m 400 /golem/work/in.hash --skip {skip} --limit {limit} {args.mask} -o /golem/work/hashcat.potfile",
+            )
+            output_file = f"hashcat_{skip}.potfile"
+            ctx.download_file(f"/golem/work/hashcat.potfile", output_file)
+            yield ctx.commit()
+            task.accept_task(result=output_file)
+
+    # beginning of the main flow
+
+    write_hash(args.hash)
+    write_keyspace_check_script(args.mask)
+
+    # By passing `event_emitter=log_summary()` we enable summary logging.
+    # See the documentation of the `yapapi.log` module on how to set
+    # the level of detail and format of the logged information.
+    async with Engine(
+        package=package,
+        max_workers=args.number_of_providers,
+        budget=10.0,
+        # timeout should be keyspace / number of providers dependent
+        timeout=timedelta(minutes=25),
+        subnet_tag=args.subnet_tag,
+        event_emitter=log_summary(log_event_repr),
+    ) as engine:
+
+        async for task in engine.map(worker_check_keyspace, [Task(data=None)]):
+            pass
+
+        keyspace = read_keyspace()
+
+        print(
+            f"{utils.TEXT_COLOR_CYAN}"
+            f"Task computed: keyspace size count. The keyspace is {keyspace}"
+            f"{utils.TEXT_COLOR_DEFAULT}"
+        )
+
+        step = int(keyspace / args.number_of_providers) + 1
+
+        ranges: range = range(0, keyspace, step)
+
+        async for task in engine.map(worker_find_password, [Task(data=range) for range in ranges]):
+            print(
+                f"{utils.TEXT_COLOR_CYAN}"
+                f"Task computed: {task}, result: {task.output}"
+                f"{utils.TEXT_COLOR_DEFAULT}"
+            )
+
+        password = read_password(ranges)
+
+        if password == None:
+            print(f"{utils.TEXT_COLOR_RED}" "No password found" f"{utils.TEXT_COLOR_DEFAULT}")
+        else:
+            print(
+                f"{utils.TEXT_COLOR_GREEN}"
+                f"Password found: {password}"
+                f"{utils.TEXT_COLOR_DEFAULT}"
+            )
+
+
+if __name__ == "__main__":
+    import pathlib
+    import sys
+
+    parser = utils.build_parser("yacat")
+
+    parser.add_argument("--numberOfProviders", dest="number_of_providers", type=int, default=3)
```

Comment:
`--number-of-providers` would better match other parameter names (`--subnet-tag`, `--log-file`). I'd shorten that to `--num-providers` or `--max-providers`.